client/src/includes/tabs.js

```diff
@@ -0,0 +1,371 @@
+    tab.setAttribute('tabindex', '-1');
+  }
+
+  /**
+   * @param {HTMLElement}tab
+   */
+  selectTab(tab) {
+    if (tab) {
+      const tabContentId = tab.getAttribute('aria-controls');
+
+      // Unselect currently active tab
+      if (tabContentId) {
+        this.unSelectActiveTab(tabContentId);
+      }
+
+      this.state.activeTabID = tabContentId;
+
+      tab.setAttribute('aria-selected', true);
+      tab.removeAttribute('tabindex');
+      const tabContent = this.tabContainer.querySelector(`#${tabContentId}`);
+
+      if (this.animate) {
+        this.animateIn(tabContent);
+      } else {
+        tabContent.hidden = false;
+      }
+
+      // Dispatch tab selected event for the rest of the admin to hook into if needed
+      // Trigger tab specific switch event
+      this.tabList.dispatchEvent(
+        new CustomEvent('switch', { detail: { tab: tab.dataset.tab } }),
+      );
+      // Dispatch tab-changed event on the document
+      document.dispatchEvent(new CustomEvent('tab-changed'));
+
+      // Set url hash
+      if (!this.disableUrl) {
+        this.setUrlHash(tab.getAttribute('href'));
+      }
+    }
+  }
+
+  /**
+   * Fade Up and In animation
+   * @param tabContent{HTMLElement}
+   */
+  animateIn(tabContent) {
+    setTimeout(() => {
+      // eslint-disable-next-line no-param-reassign
+      tabContent.hidden = false;
+      // Wait for hidden attribute to be applied then fade in
+      setTimeout(() => {
+        tabContent.classList.add(this.state.css.animate);
+      }, this.state.transition);
+    }, this.state.transition);
+  }
+
+  /**
+   * Fade Down and Out by removing css class
+   * @param tabContent{HTMLElement}
+   */
+  animateOut(tabContent) {
+    // Wait element to transition out and then hide with hidden
+    tabContent.classList.remove(this.state.css.animate);
+    setTimeout(() => {
+      // eslint-disable-next-line no-param-reassign
+      tabContent.hidden = true;
+    }, this.state.transition);
+  }
+
+  bindEvents() {
+    if (!this.tabButtons) {
+      return;
+    }
+
+    this.tabButtons.forEach((tab, index) => {
+      tab.addEventListener('click', (e) => {
+        e.preventDefault();
+        this.selectTab(tab);
+      });
+      tab.addEventListener('keydown', this.keydownEventListener.bind(this));
+      tab.addEventListener('keyup', this.keyupEventListener.bind(this));
+      // Set index of tab used in keyboard controls
+      // eslint-disable-next-line no-param-reassign
+      tab.index = index;
+    });
+  }
+
+  /**
+   *  Handle keydown on tabs
+   * @param {Event}event
+   */
+  keydownEventListener(event) {
+    const keyPressed = event.key;
+    const { keys } = this.state;
+
+    switch (keyPressed) {
+      case keys.end:
+        event.preventDefault();
+        // Activate last tab
+        this.focusLastTab();
+        break;
+      case keys.home:
+        event.preventDefault();
+        // Activate first tab
+        this.focusFirstTab();
+        break;
```

Comment:
These ones are in key down to avoid page scroll (so you can preventDefault before the default page scroll happens). I did move the others into keydown so they can all be one big happy family of keys now 😄